src/shapes/Textbox.ts

```diff
@@ -257,15 +265,57 @@ export class Textbox extends IText {
    * @returns {Array} Array of lines
    */
   _wrapText(lines: Array<any>, desiredWidth: number): Array<any> {
-    const wrapped = [];
     this.isWrapping = true;
+    // extract all the width of all the words to optimally wrap text
+    const wordsData = this.measureWords(lines);
+    const wrapped = [];
     for (let i = 0; i < lines.length; i++) {
-      wrapped.push(...this._wrapLine(lines[i], i, desiredWidth));
+      wrapped.push(...this._wrapLine(lines[i], i, desiredWidth, wordsData));
     }
     this.isWrapping = false;
     return wrapped;
   }
 
+  /**
+   * For each line of text terminated by an hard line stop,
+   * measure each word width and extract the largest word from all.
+   * @param {string[]} lines the lines we need to measure
+   */
+  measureWords(lines: string[]): WordsWidthData {
+    const splitByGrapheme = this.splitByGrapheme,
+      data = [],
+      infix = splitByGrapheme ? '' : ' ';
+
+    let largestWordWidth = 0;
+
+    for (let i = 0; i < lines.length; i++) {
+      let offset = 0;
+      const words = splitByGrapheme
+        ? this.graphemeSplit(lines[i])
+        : this.wordSplit(lines[i]);
+
+      // fix a difference between split and graphemeSplit
+      if (words.length === 0) {
+        words.push([]);
+      }
+
+      // measure words
+      data[i] = words.map((word) => {
```

Comment:
why not use lines.map?

Reply:
welll the previus for loop didn't exist, so i can use lines.map yes